app_dart/lib/src/request_handlers/refresh_github_commits.dart

```diff
@@ -61,8 +61,21 @@ class RefreshGithubCommits extends ApiRequestHandler<Body> {
         config, branchHttpClientProvider, log, gitHubBackoffCalculator);
 
     for (Branch branch in branches) {
+      int hours = config.existingBranchHours;
+
+      if (branch.name != 'master') {
+        final Set<Commit> commits = await datastore
+            .queryRecentCommits(limit: 1, branch: branch.name)
+            .toSet();
+
+        /// Set [hours] to be longer if a new branch is found.
+        if (commits.isEmpty) {
+          hours = config.newBranchHours;
+        }
+      }
+
       final List<RepositoryCommit> commits =
-          await githubService.listCommits(slug, branch.name);
+          await githubService.listCommits(slug, branch.name, hours, config);
```

Comment:
Using hours may lead to a race condition e.g the cocoon backend goes down for 1:15 mins then when it gets active it will just query 1 hour window not processing the commits that happened between current_time - 1 hour 15 mins to current_time - 1 hour.

Reply:
Thanks for pointing this out. Days may sound more reasonable.